packages/contracts-bedrock/tasks/check-l2.ts

```diff
@@ -0,0 +1,628 @@
+    const GovernanceToken = await hre.ethers.getContractAt(
+      'GovernanceToken',
+      predeploys.GovernanceToken
+    )
+
+    const name = await GovernanceToken.name()
+    if (name !== 'Optimism') {
+      throw new Error('name misconfigured')
+    }
+    console.log(`  - name: ${name}`)
+
+    const symbol = await GovernanceToken.symbol()
+    if (symbol !== 'OP') {
+      throw new Error('symbol misconfigured')
+    }
+    console.log(`  - symbol: ${symbol}`)
+
+    const owner = await GovernanceToken.owner()
+    logLoud()
+    console.log(`  - owner: ${owner}`)
+    logLoud()
+
+    const totalSupply = await GovernanceToken.totalSupply()
+    console.log(`  - totalSupply: ${totalSupply}`)
+
+    await checkProxy(hre, 'GovernanceToken')
+    // No proxy at this address, don't call assertProxy
+  },
+  // L2ERC721Bridge
+  // - check version
+  L2ERC721Bridge: async (hre: HardhatRuntimeEnvironment) => {
+    const L2ERC721Bridge = await hre.ethers.getContractAt(
+      'L2ERC721Bridge',
+      predeploys.L2ERC721Bridge
+    )
+
+    const version = await L2ERC721Bridge.version()
+    assertSemver(version, 'L2ERC721Bridge')
+
+    const MESSENGER = await L2ERC721Bridge.MESSENGER()
+    if (MESSENGER === hre.ethers.constants.AddressZero) {
+      throw new Error('MESSENGER misconfigured')
+    }
+    console.log(`  - MESSENGER: ${MESSENGER}`)
+
+    const OTHER_BRIDGE = await L2ERC721Bridge.OTHER_BRIDGE()
+    if (OTHER_BRIDGE === hre.ethers.constants.AddressZero) {
+      throw new Error('OTHER_BRIDGE misconfigured')
+    }
+    logLoud()
+    console.log(`  - OTHER_BRIDGE: ${OTHER_BRIDGE}`)
+    logLoud()
+
+    await checkProxy(hre, 'L2ERC721Bridge')
+    await assertProxy(hre, 'L2ERC721Bridge')
+  },
+  // OptimismMintableERC721Factory
+  // - check version
+  OptimismMintableERC721Factory: async (hre: HardhatRuntimeEnvironment) => {
+    const OptimismMintableERC721Factory = await hre.ethers.getContractAt(
+      'OptimismMintableERC721Factory',
+      predeploys.OptimismMintableERC721Factory
+    )
+
+    const version = await OptimismMintableERC721Factory.version()
+    assertSemver(version, 'OptimismMintableERC721Factory', '1.0.0')
+
+    const BRIDGE = await OptimismMintableERC721Factory.BRIDGE()
+    console.log(`  - BRIDGE: ${BRIDGE}`)
+
+    const REMOTE_CHAIN_ID =
+      await OptimismMintableERC721Factory.REMOTE_CHAIN_ID()
+    console.log(`  - REMOTE_CHAIN_ID: ${REMOTE_CHAIN_ID}`)
+
+    await checkProxy(hre, 'OptimismMintableERC721Factory')
+    await assertProxy(hre, 'OptimismMintableERC721Factory')
+  },
+  // ProxyAdmin
+  // - check owner
+  ProxyAdmin: async (hre: HardhatRuntimeEnvironment) => {
+    const ProxyAdmin = await hre.ethers.getContractAt(
+      'ProxyAdmin',
+      predeploys.ProxyAdmin
+    )
+
+    const owner = await ProxyAdmin.owner()
+    if (owner === hre.ethers.constants.AddressZero) {
+      throw new Error('misconfigured owner')
+    }
+    logLoud()
```

Comment:
Nitpick, based on the way `logLoud` is being used, you could have a function `yell` that looks like:

```ts
const yell = (msg: string) => {
   logLoud()
   console.log(msg)
   logLoud()
}
```